tools/ping_enumerator.py

```diff
@@ -3,76 +3,141 @@
 """Scan serial ports for ping devices
     Symlinks to detected devices are created under /dev/serial/ping/
     This script needs root permission to create the symlinks
-    Exits with the number of ping devices detected
 """
+from __future__ import print_function
 import subprocess
-from brping import Ping1D
+import traceback
+import sys
+from brping import ping1d
```

Comment:
I see your line of thought in this patch, it makes sense.

This script can be made to use only the `pingmessage` module.

This will allow us to move forward without ping360 device support, and will result in less code.

No device initialization needs to be performed for any device. The only serial communication that needs to happen in this script is requests for `device_information`.

Reply:
I think that will need some extra code for the serial and implementing the baud change for the ping360. I'll give it a shot.

Reply:
I ended using the PingDevice class so I could use some of the serial things which were already implemented there.